### DISCUSSION TITLE
Merging smaller pictures into a bigger one (width kept, depth more or less sum of all the images)

### DISCUSSION THREAD

**Discussion by axlgit** [libvips/libvips#2304] [Q&A]
Hi,

I have around 150 images whose width is always the same and have taken by scrolling a long HTML page. The HTML is not on the Internet but available as manual in a seperate system. The images (1080x1920 - portrait) where taken as a video capture on another system. Unfortunately each page doesn't start at the top of the image and have just the subsequent page as an exceeding area, it would have been much easier in such case, because knowing the area of page in WxH pixels it would be as easy as cropping the op of the page.

Hence, which approach would you suggest to merge all the 150 pages (1080 wide and roughly 1920x150 long) into a very long image withouth any overlap or discontinuity between the pages?

Then the HTML page does have a blue area between pages and this blue areas be the separator between pages, after getting this long long long image in order to split pages it will be sufficient to crop in between but this is a second step.

For the first step how do I proceed? I guess that instead of comparing line by line I should compare small areas because easily the to-be-compared line would be made by just white pixels and this is quite common in page. So I though of an area as deep as half a centimenter (value that has to be converted to pixels of course) or a centimer.

This might be a feature request too :-)

What would you think?

Alex

EDIT 0:
basically the first step is the same as merging images taken from a smartphone chat, where images where taken while scrolling the chat.
The final image will be a long image with the whole chat.
The second step is a bit clearer to me but advices will be very grateful.

**Comment by jcupitt** [comment]
Hi @axlgit,

You need `mosaic`. It'll search an overlap for a match and merge a pair of images. 

Can you post a pair of sample images?
